backend/src/test/java/codezap/auth/configuration/AuthArgumentResolverTest.java

```diff
@@ -0,0 +1,168 @@
+                new StandardServletAsyncWebRequest(httpServletRequest, new MockHttpServletResponse());
+        Member member = MemberFixture.getFirstMember();
+        Cookie credentialCookie = new Cookie(CREDENTIAL_COOKIE_NAME, credentialProvider.createCredential(member));
+
+        @Nested
+        @DisplayName("required 값이 false 라면")
+        class RequiredFalseTest {
+            Method notRequiredMethod = ReflectionSupport.findMethod(ResolveTestController.class,
+                            "notRequiredMethod", Member.class)
+                    .get();
+
+            @Test
+            @DisplayName("credential 정보가 없을때 null 이 반환된다.")
+            void noCredentialTest() {
+                //given
+
+                //when
+                Member member = resolveArgument(notRequiredMethod, nativeWebRequest);
+
+                //then
+                assertThat(member).isNull();
+            }
+
+            @Test
+            @DisplayName("credential 정보가 존재하면 Member 가 반환된다.")
+            void existCredentialTest() {
+                //given
+                httpServletRequest.setCookies(credentialCookie);
+
+                //when
+                Member resolvedArgument = resolveArgument(notRequiredMethod, nativeWebRequest);
+
+                //then
+                assertThat(resolvedArgument).isEqualTo(member);
+            }
+        }
+
+        @Nested
+        @DisplayName("required 값이 true 라면")
+        class RequiredTrueTest {
+            Method requiredMethod = ReflectionSupport.findMethod(ResolveTestController.class,
+                            "requiredMethod", Member.class)
+                    .get();
+
+            @Test
+            @DisplayName("credential 정보가 없을때 예외가 발생한다..")
+            void noCredentialTest() {
+                //given
+
+                //when & then
+                assertThatThrownBy(() -> resolveArgument(requiredMethod, nativeWebRequest))
+                        .isInstanceOf(CodeZapException.class)
+                        .hasMessage("쿠키가 없어서 회원 정보를 찾을 수 없습니다. 다시 로그인해주세요.");
+            }
+
+            @Test
+            @DisplayName("credential 정보가 존재하면 Member 가 반환된다.")
+            void existCredentialTest() {
+                //given
+                httpServletRequest.setCookies(credentialCookie);
+
+                //when
+                Member resolvedArgument = resolveArgument(requiredMethod, nativeWebRequest);
+
+                //then
+                assertThat(resolvedArgument).isEqualTo(member);
+            }
+        }
+
+        private Member resolveArgument(Method method, NativeWebRequest webRequest) {
+            ModelAndViewContainer modelAndViewContainer = new ModelAndViewContainer();
+            WebDataBinderFactory webDataBinderFactory = (request, target, objectName) -> null;
+
+            return authArgumentResolver.resolveArgument(
+                    new MethodParameter(method, 0),
+                    modelAndViewContainer,
```

Comment:
`ModelAndViewContainer` 따로 지역 변수로 빼지 않아도 될 것 같슴다.
이유가 있었다면 말해주세요!

```suggestion
            WebDataBinderFactory webDataBinderFactory = (request, target, objectName) -> null;

            return authArgumentResolver.resolveArgument(
                    new MethodParameter(method, 0),
                    new ModelAndViewContainer(),
```

Reply:
없었습니다!!!
몰리가 제안해 준 코드 보니까 깔끔하고 좋네용~